pkgpanda/util.py

```diff
@@ -413,3 +414,74 @@ def hash_checkout(item):
         return hash_list(list(item))
     else:
         raise NotImplementedError("{} of type {}".format(item, type(item)))
+
+
+def split_by_token(token_prefix, token_suffix, string_, strip_token_decoration=False):
```

Comment:
We can't define a universal token format because this function has to detect references in AWS and Azure templates, whose syntax we don't control. See usage examples for the different token decoration we need to catch.